src/test/kotlin/com/tinder/StateMachineTest.kt

```diff
@@ -698,6 +698,30 @@ internal class StateMachineTest {
             }
         }
 
+        class WithIncompleteStateDefinition {
+
+            private val stateMachine = StateMachine.create<String, Int, Nothing> {
+                initialState(STATE_A)
+                state(STATE_A) {
+                    on(EVENT_1) {
+                        transitionTo(STATE_B)
+                    }
+                }
+                // Missing STATE_B definition.
+            }
+
+            @Test
+            fun transition_givenMissingDestinationStateDefinition_shouldThrowIllegalStateExceptionWithStateName() {
+                // Then
+                try {
```

Comment:
I've decided to use `withMessageContaining(STATE_B.javaClass.simpleName)` as in my opinion checking the full message makes test too much coupled to the production code, what do you think about it?

Reply:
Well, the sole purpose of this PR is to make the error message more clear when a state definition is missing, so it makes all the sense to me that the test actually verifies that the error has the exact expected message. 

If we don't do so, then the test would allow messages like `"Foo ${STATE_B.javaClass.simpleName}."`, or simply ${STATE_B.javaClass.simpleName}. 
We don't want that.